swift/Workflow/Sources/SubtreeManager.swift

```diff
@@ -213,7 +214,14 @@ extension WorkflowNode.SubtreeManager {
 
             let reusableSink = sinkStore.findOrCreate(actionType: Action.self)
 
+            let signpostRef = NSObject()
+
             let sink = Sink<Action> { action in
+                if #available(iOS 12.0, *) {
```

Comment:
This check can get noisy. How do you feel about creating a convenience wrapper (maybe `workflow_os_signpost`) which wraps this check and the `os_signpost` call?

Reply:
I know. 😞 I went down that road, but the APIs for `os_signpost` unfortunately don't let you do that: https://github.com/square/workflow/issues/1083#issuecomment-623675549

The message arguments in the `os_signpost` calls get translated into `CVarArg...` in Swift. Apple would need to expose one that supports passing a `va_list`; there's no way to "splat" the resulting array in the wrapper back into variadic arguments.

The only other thing I can think of is to add very specific functions (e.g. `logWorkflowStart/End`) to our wrapper.

Reply:
Makes sense, ignore this comment then!